internal/blocksync/reactor.go

```diff
@@ -186,15 +186,24 @@ func (r *Reactor) OnStop() {
 func (r *Reactor) respondToPeer(ctx context.Context, msg *bcproto.BlockRequest, peerID types.NodeID, blockSyncCh *p2p.Channel) error {
 	block := r.store.LoadBlock(msg.Height)
 	if block != nil {
+		extCommit := r.store.LoadBlockExtCommit(msg.Height)
+		if extCommit == nil {
+			r.logger.Error("peer requesting a block; we have the block but not its extended commit (%v)", block)
```

Comment:
I recommend against logging an error that is also being returned. (Often the caller will just log it again, anyway—or pass it up to someone else who will—or handle it)

(there are more of these below too)

Reply:
This is a discussion I've always had in all dev teams I've been part of.
There are typically two views on this:

(a) don't log something you also report via returning an error
(b) when in doubt, log errors like hell! Especially when not in the happy path

I have a few scars that made me evolve towards strongly supporting (b) over time. The errors paths quickly become a complex mesh of call sites and error reports. When, in practice, you are chasing a bug you can't easily repro locally, the "not logging an error you assume the caller will report" may break down (e.g., because of the bug you're chasing), and you want as much evidence as possible of which parts of the code the faulty run has hit. If there is a subtle bug somewhere you are chasing, believe me, all error logs you get seem insufficient.

This reasoning doesn't apply to panic, as you typically have the backtrace, which you usually thank god for getting in those circumstances :-)

Having said that, I'm OK if the team leans toward (a). Just wanted to report on my experience here.